pkg/provider/manager/manager.go

```diff
@@ -129,17 +133,32 @@ func (m *ProviderManager) GetProviders() map[string]Provider {
 	return providers
 }
 
-func (m *ProviderManager) RegisterProvider(pluginPath string) error {
+func (m *ProviderManager) RegisterProvider(pluginPath string, manualInstall bool) error {
 	pluginRef, err := m.initializeProvider(pluginPath)
 	if err != nil {
 		return err
 	}
 
 	m.pluginRefs[pluginRef.name] = pluginRef
 
-	p, err := m.dispenseProvider(pluginRef.client, pluginRef.name)
+	lockFilePath := filepath.Join(pluginRef.path, INITIAL_SETUP_LOCK_FILE_NAME)
+	_, err = os.Stat(lockFilePath)
+	if os.IsNotExist(err) || manualInstall {
```

Comment:
Why is `manualInstall` a special case? 

Isn't the lockfile's existence enough to check? In my opinion, even if the user manually installs the provider again, if it was setup already, no need to set it up again.

Reply:
It is made this way because installing manually via the controller (cli) should always trigger setting preset targets since that's part of the installation/registration process

Reply:
So if I were to:
1. Remove the local docker target
2. Uninstall the docker provider
3. Install it again

it's expected that the local target is set again?

That's a bit counter-intuitive since, if I restart the server after step 2., the preset (local) target would not get set.

Reply:
The thought process was that if we don't set the targets on "install" then they're not really preset targets, just "initial setup targets". 

If you restart the server after step 2, they would not get set, that is correct - since the provider doesn't exist

Reply:
Ah, right.
I'm thinking if we should set the preset targets even on manual install (if the user removed them before). I guess it can't hurt. Okay, lets leave it as is then. Thanks for explaining.